app/v1/controllers/PaymentsController.scala

```diff
@@ -0,0 +1,88 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers
+
+import cats.data.EitherT
+import cats.implicits._
+import javax.inject.{Inject, Singleton}
+import play.api.http.MimeTypes
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, ControllerComponents, Result}
+import utils.{EndpointLogContext, Logging}
+import v1.controllers.requestParsers.PaymentsRequestParser
+import v1.models.errors._
+import v1.models.request.payments.PaymentsRawData
+import v1.services.{EnrolmentsAuthService, PaymentsService}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class PaymentsController @Inject()(val authService: EnrolmentsAuthService,
+                                   requestParser: PaymentsRequestParser,
+                                   service: PaymentsService,
+                                   cc: ControllerComponents)(implicit ec: ExecutionContext)
+extends AuthorisedController(cc) with BaseController with Logging {
+
+  implicit val endpointLogContext: EndpointLogContext =
+    EndpointLogContext(
+      controllerName = "FinancialDataResource",
+      endpointName = "retrievePayments"
+    )
+
+  def retrievePayments(vrn: String, from: Option[String], to: Option[String]): Action[AnyContent] =
+    authorisedAction(vrn).async{ implicit request =>
+      logger.info(s"[PaymentsController][retrievePayments] Successfully retrieved Payments from DES")
+
+      val rawRequest: PaymentsRawData =
+        PaymentsRawData(
+          vrn = vrn,
+          from = from,
+          to = to
+        )
+
+      val result =
+        for {
+          parsedRequest <- EitherT.fromEither[Future](requestParser.parseRequest(rawRequest))
+          serviceResponse <- EitherT(service.retrievePayments(parsedRequest))
+        } yield {
+          logger.debug(s"[PaymentsController][retrievePayments] Successfully retrieved Payments from DES")
+
+          Ok(Json.toJson(serviceResponse.responseData))
+            .withApiHeaders(serviceResponse.correlationId)
+            .as(MimeTypes.JSON)
+        }
+
+      result.leftMap{errorWrapper =>
+        val correlationId = getCorrelationId(errorWrapper)
+        val result = errorResult(errorWrapper).withApiHeaders(correlationId)
+        result
+      }.merge
+
+    }
+
+  private def errorResult(errorWrapper: ErrorWrapper): Result = {
+    (errorWrapper.error: @unchecked) match {
+      case VrnFormatError | VrnFormatErrorDes |
+           FinancialDataInvalidDateFromError | InvalidDateFromErrorDes |
+           FinancialDataInvalidDateToError | InvalidDateToErrorDes |
+           FinancialDataInvalidDateRangeError | InvalidDataError
+      => BadRequest(Json.toJson(errorWrapper))
+      case LegacyNotFoundError => NotFound(Json.toJson(errorWrapper))
```

Comment:
The NOT_FOUND error which payments and liabilities throw is different to the one we normally used (called NotFoundError). The actual NOT_FOUND error these endpoints throw is called LegacyNotFoundError and has slightly different wording. It isn't necessary to include NotFoundError here since the endpoint cannot return it.

Reply:
Well, at least it shouldn't

Reply:
@yangwong9664: no NotFoundError mapped in payment... better check with the spec not with Liabilities

Reply:
Not required for Liabilities also... before the service error mapping in Liabilities is wrong. I think now it is updated. Can remove from Liabilities also